File: coffeescript/reverse-words.js
var reverse_words, reverse_words_manually;

reverse_words = function(sentence) {
    return sentence.split(' ').reverse.join(' ');
};

reverse_words_manually = function(sentence) {
    var curr_word, reversed;
    curr_word = '';
    reversed = (sentence.split('').reduce(function(reversed_sentence, char) {
        if (char === ' ') {
            reversed_sentence.unshift(curr_word);
            curr_word = '';
        } else {
            curr_word += char;
        }
        return reversed_sentence;
    }, [])).join(' ');
    return curr_word + ' ' + reversed;
};

File: ruby/reverse-words.rb
# Create a function that will reverse the words in a sentence.

# the easy way
def reverse_words(sentence)
	sentence.split(' ').reverse.join ' '
end

# manually
def reverse_words_manually(sentence)
	curr_word = ''
	sentence.chars.reduce([]) { |reversed, char|
		if char == ' '
			reversed.unshift curr_word
			curr_word = ''
		else
			curr_word << char
		end
		reversed
	}.join(' ').prepend "#{curr_word} "
end